File: Stuff/Arrays/find_missing_element.py
#######CANT HANDLE DUPLICATE NUMBERS################

def finder(arr1,arr2):
	seen=set()
	for num in arr1:
		if num not in seen:
			seen.add(num)
	for num in seen:
		if num not in arr2:
			return str(num)+" is the missing number"
	return			
print(finder([1,2,3,4,5,6,7],[3,7,2,1,4,6]))

##########HANDLE DUPLICATE NUMBERS##################
import collections
def finder2(arr1,arr2):
	dic=collections.defaultdict(int)

	for num in arr2:
		dic[num]+=1

	for num in arr1:
		if dic[num] == 0:
			return num
		else :
			dic[num]-=1
	return			


############ XOR VERSION #########################
def finder3(arr1,arr2):
	result=0
	for num in arr1+arr2:
		result^=num
		print result
	print result

File: Stuff/Arrays/find_missing_element.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int finder(int *arr1, int *arr2){
	int n1,n2;
	n1=sizeof(arr1)/sizeof(arr1[0]);
	n2=sizeof(arr2)/sizeof(arr2[0]);
	sort(arr1,arr1+n1);
	sort(arr2,arr2+n2);

	for(int i=0 ; i<max(n1,n2) ; i++){
		if(i==max(n1,n2)-1){
			return arr1[i];
		}
		if(arr1[i]!=arr2[i]){
			return arr1[i];
		}
	}
	return 0;
}

int main(){
	int *arr1=new int[1,2,3,4,5,6,7,8];
	int *arr2=new int[3,2,4,8,6,7,1];
	cout<<finder(arr1,arr2);
	return 0;
}


///// NOT WORKING WELL ////////////////////